src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/TagBuilder.cs

```diff
@@ -195,56 +177,44 @@ public void MergeAttribute(string key, string value, bool replaceExisting)
 
         public void SetInnerText(string innerText)
         {
-            InnerHtml = _htmlEncoder.HtmlEncode(innerText);
-        }
-
-        public HtmlString ToHtmlString(TagRenderMode renderMode)
-        {
-            return new HtmlString(ToString(renderMode));
+            InnerHtml = new StringHtmlContent(innerText);
         }
```

Comment:
After changing the `InnerHtml` type, this method seems redundant. It previously existed mostly to allow adding both inner text to be encoded (this method) and inner text that does not need to be encoded (assigning to `InnerHtml`).

Wondering about restoring `string InnerHtml` but changing the class' implementation to include a `bool` field that determines how `InnerHtml` is handled when `WriteTo()` is called. This `StringHtmlContent` instance wouldn't be necessary and the public surface of `TagBuilder` would change less (with no perf cost).